docs/websites/nginx/django-uwsgi-nginx-ubuntu-14.md

```diff
@@ -0,0 +1,204 @@
+
+2.  Install Django framework:
+
+        pip install Django
+
+3.  Create new Django application with the command:
+
+        django-admin.py startproject sample
+
+    This will create a `sample` directory in your home directory.
+
+4.  Switch to application directory with `cd sample` and run the following command to initialize SQLite database:
+
+		./manage.py migrate
+
+5.  Open file `sample/settings.py` with your favorite text editor and add this line to the bottom of the file:
+
+{: .file-excerpt}
+sample/settings.py
+:   ~~~ py
+    STATIC_ROOT = os.path.join(BASE_DIR, "static/")
+    ~~~
+
+	This will configure Django to put all static assets into specified directory, which is needed when running with Nginx.
+
+6.  Run the following command to put static assets into directory mentioned above:
+
+		./manage.py collectstatic
+
+7.  Start development server to test sample application:
+
+		./manage.py runserver 0.0.0.0:8080
+
+	Visit http://example.com:8080 in your browser to confirm that application is set up correctly and working. Stop development server with **Ctrl-C**.
+
+## Configure uWSGI
+
+1.  Create directory with uWSGI configuration files:
+
+		sudo mkdir -p /etc/uwsgi/sites
+
+2.  Using `sudo` create configuration file `sample.ini` with the following contents:
+
+{: .file}
+/etc/uwsgi/sites/sample.ini
+:   ~~~ ini
+	[uwsgi]
+	project = sample
+	base = /home/django
+
+	chdir = %(base)/%(project)
+	home = %(base)/Env/%(project)
+	module = %(project).wsgi:application
+
+	master = true
+	processes = 2
+
+	socket = %(base)/%(project)/%(project).sock
+	chmod-socket = 664
+	vacuum = true
+    ~~~
+
+3.  With `sudo` create an Upstart script for uWSGI:
+
+{: .file}
+/etc/init/uwsgi.conf
+:   ~~~ conf
+	description "uWSGI"
+	start on runlevel [2345]
+	stop on runlevel [06]
+	respawn
+
+	env UWSGI=/usr/local/bin/uwsgi
+	env LOGTO=/var/log/uwsgi.log
+
+	exec $UWSGI --master --emperor /etc/uwsgi/sites --die-on-term --uid django --gid www-data --logto $LOGTO
+    ~~~
+
+This script will start uWSGI in "Emperor" mode.  In this mode it will monitor `/etc/uwsgi/site` directory and will spawn instances (vassals) for each configuration file it finds. Whenever a config file is changed, the emperor will automatically restart the vassal.
+
+4. Start `uwsgi` service:
+
+		sudo service uwsgi start
+
+## Configure Nginx
+
+1.  Create Nginx site configuration file `/etc/nginx/sites-available/sample` using `sudo`:
+
+{: .file}
+/etc/nginx/sites-available/sample
+:   ~~~ conf
+	server {
+		listen 80;
+		server_name example.com;
+
+	    location = /favicon.ico { access_log off; log_not_found off; }
+		location /static/ {
+			root /home/user/sample;
+		}
+
+        location / {
+			include         uwsgi_params;
+			uwsgi_pass      unix:/home/django/sample/sample.sock;
+		}
+	}
+	~~~
+
+2.  Link newly created configuration file to `sites-enabled` directory to enable it:
+
+		sudo ln -s /etc/nginx/sites-available/sample /etc/nginx/sites-enabled
```

Comment:
Also at this point default nginx site config should be removed with `sudo rm /etc/nginx/sites-enabled/default`